EXOSIMS/Prototypes/TargetList.py

```diff
@@ -1862,60 +1849,32 @@ def starprop(self, sInds, currentTime, eclip=False):
                 return np.tile(r_targ, (nTimes, 1, 1))
 
-        # target star ICRS coordinates
-        coord_old = self.coords[sInds]
-        # right ascension and declination
-        ra = coord_old.ra
-        dec = coord_old.dec
-        # directions
-        p0 = np.array([-np.sin(ra), np.cos(ra), np.zeros(sInds.size)])
-        q0 = np.array(
-            [-np.sin(dec) * np.cos(ra), -np.sin(dec) * np.sin(ra), np.cos(dec)]
-        )
-        r0 = coord_old.cartesian.xyz / coord_old.distance
-        # proper motion vector
-        mu0 = p0 * self.pmra[sInds] + q0 * self.pmdec[sInds]
-        # space velocity vector
-        v = mu0 / self.parx[sInds] * u.AU + r0 * self.rv[sInds]
-        # set J2000 epoch
-        j2000 = Time(2000.0, format="jyear")
-
         # if only 1 time in currentTime
         if nTimes == 1 or nStars == 1 or nTimes == nStars:
             # target star positions vector in heliocentric equatorial frame
-            dr = v * (currentTime.mjd - j2000.mjd) * u.day
-            r_targ = (coord_old.cartesian.xyz + dr).T.to("pc")
+            coord_old = self.StarCatalog.coords
+            coord_new = coord_old.apply_space_motion(new_obstime=currentTime)
+            r_targ = coord_new.cartesian.xyz.T.to(u.pc)
 
             if eclip:
                 # transform to heliocentric true ecliptic frame
-                coord_new = SkyCoord(
-                    r_targ[:, 0],
-                    r_targ[:, 1],
-                    r_targ[:, 2],
-                    representation_type="cartesian",
-                )
-                r_targ = coord_new.heliocentrictrueecliptic.cartesian.xyz.T.to("pc")
+                coord_new = coord_new.heliocentrictrueecliptic
+                r_targ = coord_new.cartesian.xyz.T.to(u.pc)
             return r_targ
 
         # create multi-dimensional array for r_targ
         else:
             # target star positions vector in heliocentric equatorial frame
             r_targ = np.zeros([nTimes, nStars, 3]) * u.pc
             for i, m in enumerate(currentTime):
-                dr = v * (m.mjd - j2000.mjd) * u.day
-                r_targ[i, :, :] = (coord_old.cartesian.xyz + dr).T.to("pc")
+                coord_new = coord_old.apply_space_motion(new_obstime=m)
+                r_targ[i, :, :] = coord_new.cartesian.xyz.T.to(u.pc)
 
             if eclip:
                 # transform to heliocentric true ecliptic frame
-                coord_new = SkyCoord(
-                    r_targ[i, :, 0],
-                    r_targ[i, :, 1],
-                    r_targ[i, :, 2],
-                    representation_type="cartesian",
-                )
-                r_targ[i, :, :] = coord_new.heliocentrictrueecliptic.cartesian.xyz.T.to(
-                    "pc"
-                )
+                for i, m in enumerate(currentTime):
+                    coord_new = coord_old.apply_space_motion(new_obstime=m)
+                    r_targ[i, :, :] = coord_new.cartesian.xyz.T.to(u.pc)

```

Comment:
I'm not seeing where the transformation to ecliptic coordinates is happening here?